drafts/keep-the-lights-on.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Keep the lights on
+subtitle: Design your system for failures
+domain: software-engineering-corner.hashnode.dev
+slug: blog-one-table-to-rule-them-all
+tags: web-development, microservices, software-architecture, frontend, web-components
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1703247396853/r8H7Xfeks.jpg?auto=format
+publishAs: lehmamic
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+Error handling is rather neglected in software projects and is more of a minor matter. No wonder, as it is much better to concentrate on business value and features. Unfortunately, this can be very annoying for users. Even worse, it can end in disaster.
+
+> In August 2013, Google experienced a five-minute outage, resulting in an estimated loss of $545,000 in revenue. Users were unable to perform searches or access other Google services during that time.
+
+Or a similar example.
+
+> In 2013, Amazon faced issues with its online retail system during a one-hour outage, potentially causing a loss of $66,240 per minute. The total estimated loss was around $3.85 million.
+
+As we have seen, even one single minute of downtime can cost a fortune or have legal implications.
+Yes, of course, my examples are about very big, well-known companies. You might think this does not concern you and your software projects. **You are wrong.** Most public-facing applications have a monetary or legal aspect.
+
+## Design for failure
+
+First I want to talk about the current situation. At this point, we might have realized that proper error handling is important. In reality, unfortunately, the realization usually looks very poor.
+
+We know all those little nice error popups thrown into our faces when an error has occurred in the application we are working on. Sadly, this is common practice in most applications. Exceptions bubble up to the frontend, get displayed in an error dialog and nothing works anymore. And be honest, most likely you implemented such an error dialog as well.
```

Comment:
I would remove nice here :)
Also we all know this from any application, not just the one we are working on
```suggestion
We know all those little error popups thrown into our faces when an error has occurred in the application. Sadly, this is common practice in most applications. Exceptions bubble up to the frontend, get displayed in an error dialog and nothing works anymore. And be honest, most likely you implemented such an error dialog as well.
```